tests/v2/test_jsonrpc_waku.nim

```diff
@@ -281,3 +282,64 @@ procSuite "Waku v2 JSON-RPC API":
     server.stop()
     server.close()
     waitfor node.stop()
+  
+  asyncTest "Filter API: get latest messages":
+    const cTopic = ContentTopic(1)
+
+    waitFor node.start()
+
+    # RPC server setup
+    let
+      rpcPort = Port(8545)
+      ta = initTAddress(bindIp, rpcPort)
+      server = newRpcHttpServer([ta])
+
+    installFilterApiHandlers(node, server)
+    server.start()
+    
+    node.mountFilter()
+
+    let client = newRpcHttpClient()
+    await client.connect("127.0.0.1", rpcPort)
+
+    # First ensure subscription exists
+
+    let sub = await client.post_waku_v2_filter_v1_subscription(contentFilters = @[ContentFilter(topics: @[cTopic])], topic = some(defaultTopic))
+    check:
+      sub
+
+    # Now prime the node with some messages before tests
+    var
```

Comment:
If we do this in every test it make make sense to pull out into a util function, to make tests a tad less verbose. This is fine too though